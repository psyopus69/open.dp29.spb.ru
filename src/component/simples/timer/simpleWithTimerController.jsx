import React, {Fragment} from "react";
import {PropTypes} from "prop-types";

// Complete Counter with buttons & timer
const SimpleWithTimerController = (props) => {
  const {renderTimer, renderChangeButton, renderResetButton} = props;
  return <Fragment>
    <Fragment>Test 1.1 timer React<br/>Timer:<br/> â˜º
    </Fragment>
    <Fragment>
      {renderTimer()}
    </Fragment>
    <Fragment>Controls P/S:</Fragment>
    <Fragment>
      {renderChangeButton()}
    </Fragment>
    <Fragment><br/>Reset : </Fragment>
    <Fragment>
      {renderResetButton()}
    </Fragment>
  </Fragment>;
};
SimpleWithTimerController.propTypes = {
  renderTimer: PropTypes.func.isRequired,
  renderChangeButton: PropTypes.func.isRequired,
  renderResetButton: PropTypes.func.isRequired,
};

export {
  SimpleWithTimerController,
};
