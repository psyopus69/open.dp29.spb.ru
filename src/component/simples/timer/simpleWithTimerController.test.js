import React from "react";
import renderer from "react-test-renderer";
import {SimpleWithTimerController} from "./simpleWithTimerController";

it(`panel with timer & control buttons play/pause & reset`, () => {
  const tree = renderer.create(<SimpleWithTimerController renderTimer={jest.fn()} renderChangeButton={jest.fn()} renderResetButton={jest.fn()}
  />);
  expect(tree).toMatchSnapshot();
});
