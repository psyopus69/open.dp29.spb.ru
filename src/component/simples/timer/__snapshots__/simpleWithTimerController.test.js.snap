// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel with timer & control buttons play/pause & reset 1`] = `
Array [
  "Test 1.1 timer React",
  <br />,
  "Timer:",
  <br />,
  " â˜º",
  "Controls P/S:",
  <br />,
  "Reset : ",
]
`;
